api/src/main/java/org/apache/iceberg/util/StructProjection.java

```diff
@@ -58,12 +58,32 @@ public static StructProjection create(Schema dataSchema, Schema projectedSchema)
     return new StructProjection(dataSchema.asStruct(), projectedSchema.asStruct());
   }
 
+  /**
+   * Creates a projecting wrapper for {@link StructLike} rows.
+   * <p>
+   * This projection does not work with repeated types like lists and maps.
+   *
+   * @param structType type of rows wrapped by this projection
+   * @param projectedStructType result type of the projected rows
+   * @param projectMissingFieldsAsNulls a flag whether to project missing fields as nulls
```

Comment:
You might consider renaming this to `allowMissing`. The original behavior is to fail if there is a field missing from the base type and this change really just allows that case and does the most reasonable thing -- it fills in null. But we don't necessarily want to guarantee that this will always fill in null. So I'd change the boolean to state that it allows missing fields rather than it fills in nulls.

Also, I think it is better for APIs in general to hide boolean arguments. Since this is already a factory method, why not use a name that signals missing fields are allowed? Something like `create` and `createAllowMissing`?

Reply:
Fixed.